.github/workflows/release.yml

```
@@ -1,28 +1,68 @@
 name: Build and Release
 
+env:
+  go-version: "1.19"
+
 on:
   release:
     types: [created]
 
 jobs:
-  releases-matrix:
-    name: Release Go Binary
-    runs-on: ubuntu-latest
+  release:
     strategy:
       matrix:
         # build and publish in parallel: linux/amd64, linux/arm64, windows/amd64, darwin/amd64, darwin/arm64
-        goos: [linux, windows, darwin]
-        goarch: [amd64, arm64]
-        exclude:
-          - goarch: arm64
-            goos: windows
+        include:
+          - image: macos-latest
+            name: macos-amd64
+            extension: tar.gz
+          - image: ubuntu-latest
+            name: linux-amd64
+            extension: tar.gz
+          - image: windows-latest
+            name: windows-amd64
+            extension: zip
+          - image: [self-hosted, macos, arm64]
+            name: macos-arm64
+            extension: tar.gz
+          - image: [self-hosted, linux, arm64]
+            name: linux-arm64
+            extension: tar.gz
+    env:
+      BINARY_NAME: smcli
+      ARTIFACT_NAME: smcli-${{ github.event.release.tag_name }}-${{ matrix.name }}.${{ matrix.extension }}
+    runs-on: ${{ matrix.image }}
+    name: Release ${{ matrix.name }}
     steps:
-    - uses: actions/checkout@v3
-    - uses: wangyoucao577/go-release-action@v1
+    - name: Checkout
+      uses: actions/checkout@v3
+    - name: Set up go
+      uses: actions/setup-go@v4
+      with:
+        go-version: ${{ env.go-version }}
+    - name: Install musl
+      if: contains(matrix.name, 'linux')
+      uses: awalsh128/cache-apt-pkgs-action@v1
+      with:
+        packages: musl-tools # provides musl-gcc
+        version: 1.0
+    - name: Build
+      run: make build
+    - name: Prepare files
+      if: ${{ !contains(matrix.name, 'windows') }}
+      run: |
+        mkdir artifacts
+        mv LICENSE README.md ${{ env.BINARY_NAME }} artifacts
+        cd artifacts
+        tar -czf ${{ env.ARTIFACT_NAME }} *
+        mv ${{ env.ARTIFACT_NAME }} ..
+    - name: Prepare files Windows
+      if: contains(matrix.name, 'windows')
+      run: |
+        mkdir artifacts
+        Move-Item -Path LICENSE, README.md, ${{ env.BINARY_NAME }}.exe -Destination artifacts
+        Compress-Archive -Path artifacts/* -Destination ${{ env.ARTIFACT_NAME }}
```

I think this should work without the need to check if it is a windows system or not:

```suggestion
    - name: Prepare files
      shell: bash
      run: |
        mkdir artifacts
        mv LICENSE README.md ${{ env.BINARY_NAME }} artifacts
        cd artifacts
        tar -czf ${{ env.ARTIFACT_NAME }} *
        mv ${{ env.ARTIFACT_NAME }} ..
```